internal/pkg/manifest/workload.go

```diff
@@ -624,6 +627,63 @@ func (i ImageWithHealthcheck) HealthCheckOpts() *ecs.HealthCheck {
 	return i.HealthCheck.healthCheckOpts()
 }
 
+// PlatformArgsOrString is a custom type which supports unmarshaling yaml which
+// can either be of type string or type PlatformArgs.
+type PlatformArgsOrString struct {
+	PlatformString *string
+	PlatformArgs   PlatformArgs
+}
+
+// UnmarshalYAML overrides the default YAML unmarshaling logic for the PlatformArgsOrString
+// struct, allowing it to perform more complex unmarshaling behavior.
+// This method implements the yaml.Unmarshaler (v2) interface.
+func (p *PlatformArgsOrString) UnmarshalYAML(unmarshal func(interface{}) error) error {
+	if err := unmarshal(&p.PlatformArgs); err != nil {
+		switch err.(type) {
+		case *yaml.TypeError:
+			break
+		default:
+			return err
+		}
+	}
+
+	if !p.PlatformArgs.isEmpty() {
+		if !p.PlatformArgs.bothOrNeitherSpecified() {
+			return errors.New(`fields 'osfamily' and 'architecture' must either both be specified or both be empty. For more info, see your workload's manifest documentation at https://aws.github.io/copilot-cli/`)
```

Comment:
can both of these fields be `nil` if `!p.PlatformArgs.IsEmpty()`? 
I wonder if we should have instead:
```suggestion
		if !p.PlatformArgs.hasAllFieldsSet() {
			return errors.New(`fields 'osfamily' and 'architecture' must either both be specified)
```

Reply:
nit: My PoV (which I might be completely wrong 😝) is that error messages from `Error() string` should be succinct and outline the cause of the error.  

In this situation, since you already give such a nice err message we can remove the recommendation.

Instead, we can have a client like the `cli` pkg catch the err and decide to log additional info like the documentation link. Or even better have our errors implement an interface like `Recommendation() string` to give these links.

Reply:
I thought checking for both and neither would make it more universal if it ever isn't within `!p.PlatformArgs.IsEmpty()`. But I guess this is the only usage of it. Will change! 

👍🏼  on error message, too.